lib/chef/provider/git.rb

```diff
@@ -282,49 +312,58 @@ def rev_search_pattern
       end
 
       def git_ls_remote(rev_pattern)
-        command = git(%Q{ls-remote "#{@new_resource.repository}" "#{rev_pattern}"})
-        shell_out!(command, run_options).stdout
+        stdout_obj = git_standard_executor ["ls-remote", "\"#{@new_resource.repository}\"", "\"#{rev_pattern}\""]
```

Comment:
you should be able to make this look like `git_standard_executor("ls-remote", new_resource.repository, rev_pattern)` and eliminate all the interpolation